### PR TITLE
symbols: Fix columns

### PR DESCRIPTION
**Pull request by chrismwendt**
**Before**

image

**After**

CleanShot 2022-03-07 at 19 13 18@2x

- Made the symbols service return lines 0-based
- Added `character` to the output of the symbols service
- Adjusted back to 1-based in the `frontend/backend` package (lol it really is called that)
- Dropped `pattern`

Resolves https://github.com/sourcegraph/sourcegraph/issues/32295

Depends on https://github.com/sourcegraph/sourcegraph/pull/32300

There's another issue that seems relevant https://github.com/sourcegraph/sourcegraph/pull/23664 @rvantonder am I missing something here?

The indexed search query path still seems to work:

CleanShot 2022-03-07 at 19 32 37

## Test plan

Ran locally, also added a test.

### CODE REVIEW

File: enterprise/cmd/symbols/main.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func createParserWithConfig(config types.CtagsConfig) rockskip.ParseSymbolsFunc
 				Name:   entry.Name,
 				Parent: entry.Parent,
 				Kind:   entry.Kind,
-				Line:   entry.Line,
+				Line:   entry.Line - 1,
```
Comment (efritz) [inline enterprise/cmd/symbols/main.go:121]: this hurts my bones

File: cmd/frontend/backend/symbols.go [hunk 2]
```diff
@@ -15,5 +15,12 @@ type symbols struct{}
 
 // ListTags returns symbols in a repository from ctags.
 func (symbols) ListTags(ctx context.Context, args search.SymbolsParameters) (result.Symbols, error) {
-	return symbolsclient.DefaultClient.Search(ctx, args)
+	symbols, err := symbolsclient.DefaultClient.Search(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+	for i := range symbols {
+		symbols[i].Line += 1 // callers expect 1-indexed lines
```
Comment (efritz) [inline cmd/frontend/backend/symbols.go:23]: oof ouch right in my bones

**Review by efritz — APPROVED** [review]
Halfway through the review I realized the implication and I LIKE IT.

**Review by rvantonder — COMMENTED** [review]
> There's another issue that seems relevant https://github.com/sourcegraph/sourcegraph/pull/23664 @rvantonder am I missing something here?

Looks good, and looks like you actually helped to resolve some of the issues with using `pattern`. Honestly though, I don't remember much of the code here (full cache eviction, it's been a long time 😆) and I currently consider this code up till now basically ownerless (in a "do what you want" kind of way).


nit:

> Passing -1 as the character will make NewSymbolMatch infer the column from the line and symbol name.

Looks like [all callers call this with `-1`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e6dd3472eb6a55c1e89c7ac65c3e0cecb725eb24/-/blob/internal/search/result/symbol.go?L34:6#tab=references) so maybe just don't pass it and always run that code?

**Comment by chrismwendt** [comment]
"full cache eviction" - heh completely understandable 😅 

I'll leave the `-1` in there as breadcrumbs for the next engineer (or myself lol) if/when they add column info.

**Pull request merged**